src/node_buffer.cc

```diff
@@ -565,6 +566,53 @@ void StringSlice(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(ret);
 }
 
+// Convert the input into an encoded string
+void DecodeUTF8(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);  // list, flags
+
+  if (!(args[0]->IsArrayBuffer() || args[0]->IsSharedArrayBuffer() ||
+        args[0]->IsArrayBufferView())) {
+    return node::THROW_ERR_INVALID_ARG_TYPE(
+        env->isolate(),
+        "The \"list\" argument must be an instance of SharedArrayBuffer, "
+        "ArrayBuffer or ArrayBufferView.");
+  }
+
+  ArrayBufferViewContents<char> buffer(args[0]);
+
+  CHECK(args[1]->IsBoolean());
+  bool ignore_bom = args[1]->BooleanValue(env->isolate());
+
+  const char* data = buffer.data();
+  size_t beginning = 0;
+  size_t length = buffer.length();
+
+  if (length == 0) return args.GetReturnValue().SetEmptyString();
+
+  if (!ignore_bom && length >= 3) {
+    char bom[] = "\xEF\xBB\xBF";
+
+    if (memcmp(data, bom, 3) == 0) {
+      beginning += 3;
+      length -= 3;
+    }
+  }
+
+  auto output = data + beginning;
```

Comment:
You can get rid of both `output` and `beginning` if you change `beginning += 3` to `data += 3`.

Fewer variables in general makes code easier to read (fewer things to hold in your head.)

Reply:
Really good recommendation. Thanks!